lib/envs.ts

```diff
@@ -0,0 +1,18 @@
+type Environment = Record<string, string>
+
+/**
+ * This is an alternative to using `process.env` directly. By using the exported proxy,
+ * `env.BUCKET_NAME` will throw an Error if BUCKET_NAME is not defined in the environment.
+ * This eliminates the need to check the existence of each environment variable where it is used.
+ */
+const envProxy: Environment = new Proxy({}, {
+  get (_target: Record<string, string>, name: string): string {
+    const value = process.env[name]
+    if (value === undefined) {
+      throw new Error(`Environment variable ${name} is not set`)
+    }
+    return value
+  }
+})
+
```

Comment:
Interesting approach! I haven't seen this before and I like it!

As a generic utility, you probably don't need it here, but it might be nice to have an additional helper:

```js
envProxy.getOrDefault = function(field: string, defaultValue: string): string {
  const value = process.env[name]
    if (value === undefined) {
      return default
    }
    return value
}
```

So you can easily default values when they are not mandatory:

```js
import envs from '...'

let region = envs.getOrDefault('AWS_REGION', 'eu-west-1')
```

Reply:
That's a nice pattern to add to the toolkit, thanks!

The Proxy approach is a bit unusual maybe but I like being able to do things like 

```javascript
const { HOME } = envs
```

and get the error handling for free

Reply:
Yes, it's fancy but I really liked it :)